src/hotspot/os/posix/os_posix.cpp

```diff
@@ -889,9 +889,9 @@ char* os::build_agent_function_name(const char *sym_name, const char *lib_name,
 
 // Sleep forever; naked call to OS-specific sleep; use with CAUTION
 void os::infinite_sleep() {
-  while (true) {    // sleep forever ...
+  sleep:            // sleep forever ...
     ::sleep(100);   // ... 100 seconds at a time
-  }
+  goto sleep;
```

Comment:
ah, when I was searching for functions to implement the noreturn with and stumbled across this one, I thought it could do with a goto instead of a while true since the intent seemed to be clearer. I can revert this if need be

Reply:
Please revert - `while (true)` conveys the exact intent.